test/com/mjcode/model/PieceTest.java

```diff
@@ -56,34 +56,116 @@ void testCanCapture() {
 
     @Test
     void testCanMoveToOccupiedTile() {
-        Tile targetTile = new Tile(5,5);
+        Tile targetTile = new Tile(3,4);
         Piece whitePiece = new Piece(player1, whiteCurrentTile, 1);
 
-        assertFalse(targetTile.isOccupied());
+        // Add piece to the target tile
+        targetTile.setOccupant(whitePiece);
```

Comment:
whitePiece is on whiteCurrentTile (2,5) - targetTile (3,4) should have a different piece occupying it